src/Handler/TagHandler.php

```diff
@@ -117,9 +119,14 @@ public function addTags(array $tags)
      */
     public function invalidateTags(array $tags)
     {
-        $tagExpression = sprintf('(%s)(,.+)?$', implode('|', array_map('preg_quote', $this->escapeTags($tags))));
-        $headers = [$this->tagsHeader => $tagExpression];
-        $this->invalidator->invalidate($headers);
+        if (empty($tags)) {
+            return $this;
+        }
+
+        $this->invalidator->invalidateTags(
+            $this->escapeTags($tags),
```

Comment:
> do you plan to use a header for tagging responses with HttpCache as well?

Kind of, but I planning to ask you and @ddeboer how this can be done in a clean way.
Could also be that the ProxyClients -> drivers gets involved in generating the output.